tools/DB_date_zeros_removal.php

```diff
@@ -106,11 +106,25 @@
             " SET ".$db->escape($field['COLUMN_NAME'])."=NULL".$autoUpdateSQL.
             " WHERE CAST(".$db->escape($field['COLUMN_NAME'])." AS CHAR(20))='0000-00-00 00:00:00';\n";
     } else {
-	echo "COLUMN ".$field['COLUMN_NAME']." in TABLE ".$field['TABLE_NAME']." is NOT NULLABLE. ".
-	    "A date '1000-01-01' will be entered instead of '0000-00-00' values.\n"; 
-        $nonNullUpdates .= "UPDATE ".$db->escape($database['database']).".".$db->escape($field['TABLE_NAME']).
-            " SET ".$db->escape($field['COLUMN_NAME'])."='1000-01-01'".$autoUpdateSQL.
-            " WHERE CAST(".$db->escape($field['COLUMN_NAME'])." AS CHAR(20))='0000-00-00';\n";
+        if ($field['DATA_TYPE'] == 'date') {
+            echo "COLUMN ".$field['COLUMN_NAME']." in TABLE ".$field['TABLE_NAME']." is NOT NULLABLE. ".
+                "A date '1000-01-01' will be entered instead of '0000-00-00' values.\n";
+            $nonNullUpdates .= "UPDATE ".$db->escape($database['database']).".".$db->escape($field['TABLE_NAME']).
+                " SET ".$db->escape($field['COLUMN_NAME'])."='1000-01-01'".$autoUpdateSQL.
+                " WHERE CAST(".$db->escape($field['COLUMN_NAME'])." AS CHAR(20))='0000-00-00';\n";
+        } else if ($field['DATA_TYPE'] == 'datetime') {
+            echo "COLUMN ".$field['COLUMN_NAME']." in TABLE ".$field['TABLE_NAME']." is NOT NULLABLE. ".
+                "A date '1000-01-01' will be entered instead of '0000-00-00' values.\n";
```

Comment:
@llevitis I think the message 'A date '1000-01-01' will be entered instead of '0000-00-00' values'  should be changed to something like this; 'A datetime '1000-01-01 00:00:00' will be entered instead of '0000-00-00' values.
Otherwise the update statement and the message doesn't match each other.

Reply:
done!